src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonNuGetProject.cs

```diff
@@ -298,7 +298,7 @@ private JObject GetJson()
         {
             try
             {
-                using (var streamReader = new StreamReader(_jsonConfig.OpenRead()))
+                using (var streamReader = new StreamReader(_jsonConfig.Open(FileMode.Open, FileAccess.Read, JsonPackageSpecReader.share)))
```

Comment:
so people don't make this mistake in the future and just use the one method that's supposed to do the reading work for project.json.